### ISSUE TITLE
Compilation error with avr-gcc-7.3

### ISSUE THREAD

**Issue by jputcu** [ETLCPP/etl#704]
After updating the code to "etl/20.36.0" I get the following compilation errors:

~~~~
include\etl\parameter_pack.h:135:74: error: template definition of non-template 'constexpr const size_t etl::parameter_pack<TTypes>::index_of_type<T>::value'
   constexpr size_t parameter_pack<TTypes...>::template index_of_type<T>::value;
                                                                          ^~~~~
~~~~

**Comment by jputcu** [comment]
Removing the `::template` fixed the error.

**Comment by jputcu** [comment]
But isn't the compiler correct, my mingw gcc 13.1 doesn't complain after removing the `template`?

**Comment by jwellbelove** [comment]
I'm pretty sure the keyword is necessary according to the standard. The ETL's tests include compilation with GCC 12.

**Comment by jwellbelove** [comment]
I'll check in the Git file history to see when it was added.

**Comment by jwellbelove** [comment]
Fixed in Pull Request #707
